src/Mvc/Mvc.Core/src/Infrastructure/ProblemDetailsJsonConverter.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc.Core;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    internal class ProblemDetailsJsonConverter : JsonConverter<ProblemDetails>
+    {
+        private static readonly JsonEncodedText Type = JsonEncodedText.Encode("type");
+        private static readonly JsonEncodedText Title = JsonEncodedText.Encode("title");
+        private static readonly JsonEncodedText Status = JsonEncodedText.Encode("status");
+        private static readonly JsonEncodedText Detail = JsonEncodedText.Encode("detail");
+        private static readonly JsonEncodedText Instance = JsonEncodedText.Encode("instance");
+
+        public override ProblemDetails Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var problemDetails = new ProblemDetails();
+
+            if (!reader.Read())
+            {
+                throw new InvalidDataException(Resources.UnexpectedJsonEnd);
+            }
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                ReadValue(ref reader, problemDetails, options);
+            }
```

Comment:
It looks like this would tolerate invalid JSON that omits the closing `}`. We should check the current token after the while loop.